src/server/pps/server/api_server.go

```diff
@@ -1347,129 +1355,51 @@ func (a *apiServer) GetLogs(request *pps.GetLogsRequest, apiGetLogsServer pps.AP
 
 		// If the job had stats enabled, we use the logs from the stats
 		// commit since that's likely to yield better results.
+		statsLogs := false
 		if statsCommit != nil {
 			ci, err := pachClient.InspectCommit(statsCommit.Repo.Name, statsCommit.ID)
 			if err != nil {
 				return err
 			}
 			if ci.Finished != nil {
-				return a.getLogsFromStats(pachClient, request, apiGetLogsServer, statsCommit)
+				statsLogs = true
+				eg.Go(func() error {
```

Comment:
very very minor style nit, but how would you feel about having this be:
```
if ci.Finished != nil {
  statsLogs = true
}
...
switch {
  case statsLogs:
    eg.Go(func() error {
      return a.getLogsFromStats(pachClient, request, statsCommit, logCh)
    })
  case request.Follow:
    eg.Go(func() error {
      return a.followLogsForRC(pachClient, request, containerName, "", logCh)
    })
  default:
    eg.Go(func() error {
      return a.getLogsForRC(pachClient, request, containerName, "", logCh)
    })
}
```

Reply:
not minor at all! it definitely clarifies things, will add it.